core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/iteration/IterateListSimultaneousltUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.list.iteration;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.util.StreamUtils;
+
+public class IterateListSimultaneousltUnitTest {
+
+    List<String> names = List.of("John", "Nik", "Tom");
+    List<Integer> numbers = List.of(1, 2, 3);
+
+    @Test
+    public void givenTwoLists_whenProcessByZipping_thenProcessedSimultaneously() {
+        Stream<String> nameStream = names.stream();
+        Stream<Integer> numberStream = numbers.stream();
+        List<String> processedList = StreamUtils.zip(numberStream, nameStream,
+            (numStr, nameStr) -> String.format("[Using Zip] Processing ArrayList Simultaneously : %d, %s", numStr, nameStr))
+            .collect(Collectors.toList());
+        Assertions.assertTrue(processedList.containsAll(
```

Comment:
It's unconventional to use `Assertions.` in tests. Static import it.

This assertion is also a weak one. If it fails, you'll get `expected true, was false`. We really want to assert that the output list contains exactly the right items in the expected order. I suggest you use `AssertJ` and `assertThat(processedList).containsExactly(...)`